appservice/test/maskValue.test.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { maskValue } from '../src/utils/maskValue';
+
+suite("maskValue Tests", () => {
+    const credentials: string = 'scHQERrAlXSmlCeN1mrhDzsHWeDz2XZt5R343HgCNmxS0xlswcaA2Cowflda';
+    suite("maskValue", () => {
+        test("Value masked (single occurance)", async () => {
+
+            // tslint:disable-next-line: no-multiline-string
+            let errorMessage: string = `To https://naturins-22-error-03.scm.azurewebsites.net:443/naturins-22-error-03.git
+            ! [rejected]        HEAD -> master (fetch first)
+           error: failed to push some refs to 'https://$naturins-22-error-03:scHQERrAlXSmlCeN1mrhDzsHWeDz2XZt5R343HgCNmxS0xlswcaA2Cowflda@naturins-22-error-03.scm.azurewebsites.net:443/naturins-22-error-03.git'
+           hint: Updates were rejected because the remote contains work that you do
+           hint: not have locally. This is usually caused by another repository pushing
+           hint: to the same ref. You may want to first integrate the remote changes
+           hint: (e.g., 'git pull ...') before pushing again.
+           hint: See the 'Note about fast-forwards' in 'git push --help' for details.`;
+
+            errorMessage = maskValue(errorMessage, credentials);
```

Comment:
These unit tests are a good start, but they don't quite cover the scenario Stephen brought up (where you forgot to reassign the result after masking the value). Instead of just `maskValue`, I would create a function like this so that the tests are more valuable
```
export async function callWithMaskHandling<T>(valueToMask: string, callback: () => Promise<T>): Promise<T> {
    try {
        return await callback();
    } catch (error) {
        // mask logic
    }
}
```